### PR TITLE
Update trial info

### LINKED ISSUE #2907: Update Trial docs about deletion
### Issue Type

Inaccurate, misleading, or out-of-date information

### Links to Affected Docs

https://docs.astronomer.io/astro/trial#after-your-trial

### The Issue or Context

Docs say there is a 7 day grace period to recover data/ deployments.

### Fix or Proposed Change

Add the Hosted specific info to the existing start a trial doc.

### Additional Notes
- Slack thread: https://astronomer.slack.com/archives/C01UA03TPV1/p1696286609279489

### Required Reviewers

_No response_

### PR DESCRIPTION
**Pull request by lzdanski**
Resolves #2907

### CODE REVIEW

File: astro/trial.md [hunk 1]
```diff
@@ -29,9 +29,9 @@ You have 14 days and $300 to spend in credits before your trial ends. See [Manag
...
 ## After your trial
 
-After your 14-day trial ends, you can no longer access your Deployments and Workspaces from the Cloud UI. You can still access your user account page and Astronomer support forms. Any DAGs you deployed will continue to run for an additional 7-day grace period.
+After your 14-day trial ends, you can no longer access your Deployments and Workspaces from the Cloud UI. You can still access your user account page and Astronomer support forms.
 
-After the 7-day grace period, your cluster and all Deployments within it are automatically deleted. Any code that you deployed to Astro will be lost. If you need additional time to evaluate Astro, or you need to copy your configuration for future use, you can:
+After your trial ends, your cluster and all Deployments within it are automatically deleted. Any code that you deployed to Astro will be lost. If you need additional time to evaluate Astro, or you need to copy your configuration for future use, you can:
 
 - Schedule a call with [sales](https://astronomer.io/contact/).
 - Open the Cloud UI and schedule a 15 minute call with an Astronomer engineer. This option is available only after your 14-day trial ends.
```
Comment (paolaperaza) [inline astro/trial.md:37]: Is this true?

File: astro/trial.md [hunk 2]
```diff
@@ -29,9 +29,9 @@ You have 14 days and $300 to spend in credits before your trial ends. See [Manag
 
 ## After your trial
 
-After your 14-day trial ends, you can no longer access your Deployments and Workspaces from the Cloud UI. You can still access your user account page and Astronomer support forms. Any DAGs you deployed will continue to run for an additional 7-day grace period.
+After your 14-day trial ends, you can no longer access your Deployments and Workspaces from the Cloud UI. You can still access your user account page and Astronomer support forms.
 
-After the 7-day grace period, your cluster and all Deployments within it are automatically deleted. Any code that you deployed to Astro will be lost. If you need additional time to evaluate Astro, or you need to copy your configuration for future use, you can:
+After your trial ends, your cluster and all Deployments within it are automatically deleted. Any code that you deployed to Astro will be lost. If you need additional time to evaluate Astro, or you need to copy your configuration for future use, you can:
```
Comment (paolaperaza) [inline astro/trial.md:34]:
```suggestion
After your trial ends, all Deployments are automatically deleted. If you created a dedicated cluster, it is also deleted. Any code that you deployed to Astro will be lost. If you need additional time to evaluate Astro, or you need to copy your configuration for future use, you can:
```

File: astro/trial.md (hunk 1 above)
Comment (paolaperaza) [inline astro/trial.md:37]:
```suggestion
- Contact [sales](https://astronomer.io/contact/) to request a trial extension.
- Run `astro deployment inspect` with the Astro CLI to save your existing Deployment configuration as a JSON or YAML file. See [Astro CLI command reference](cli/astro-deployment-inspect.md).
- Open the Cloud UI and schedule office hours with an Astronomer engineer.
```

**Review by paolaperaza — APPROVED** [review]

File: astro/trial.md (hunk 1 above)
Comment (lzdanski) [reply astro/trial.md:37]: I can also get Nick to do a review pass, but that's how I understood this Slack explanation - https://astronomer.slack.com/archives/C01UA03TPV1/p1696301864953699?thread_ts=1696286609.279489&cid=C01UA03TPV1

File: astro/trial.md [hunk 3]
```diff
@@ -29,9 +29,10 @@ You have 14 days and $300 to spend in credits before your trial ends. See [Manag
...
 
-After the 7-day grace period, your cluster and all Deployments within it are automatically deleted. Any code that you deployed to Astro will be lost. If you need additional time to evaluate Astro, or you need to copy your configuration for future use, you can:
+After your trial ends, all Deployments are automatically deleted. If you created a dedicated cluster, it is also deleted. Any code that you deployed to Astro will be lost. If you need additional time to evaluate Astro, or you need to copy your configuration for future use, you can:
 
-- Schedule a call with [sales](https://astronomer.io/contact/).
-- Open the Cloud UI and schedule a 15 minute call with an Astronomer engineer. This option is available only after your 14-day trial ends.
\ No newline at end of file
+- Contact [sales](https://astronomer.io/contact/) to request a trial extension.
+- Run `astro deployment inspect` with the Astro CLI to save your existing Deployment configuration as a JSON or YAML file. See [Astro CLI command reference](cli/astro-deployment-inspect.md).
+- Open the Cloud UI and schedule office hours with an Astronomer engineer.
```
Comment (lzdanski) [inline astro/trial.md:38]:
```suggestion
- Open the Cloud UI and [schedule office hours](office-hours.md) with an Astronomer engineer.
```

**Pull request merged**